crates/hotshot/src/block_impl.rs

```diff
@@ -12,30 +11,17 @@ use snafu::Snafu;
 
 /// The transaction in a [`VIDBlockPayload`].
 #[derive(PartialEq, Eq, Hash, Serialize, Deserialize, Clone, Debug)]
-pub struct VIDTransaction {
-    /// identifier for the transaction
-    pub id: u64,
-    /// padding to add to txn (to make it larger and thereby more realistic)
-    pub padding: Vec<u8>,
-}
-
-impl Deref for VIDTransaction {
-    type Target = u64;
-
-    fn deref(&self) -> &Self::Target {
-        &self.id
-    }
-}
+pub struct VIDTransaction(pub Vec<u8>);
 
 impl Committable for VIDTransaction {
     fn commit(&self) -> Commitment<Self> {
-        commit::RawCommitmentBuilder::new("SDemo Txn Comm")
-            .u64_field("id", self.id)
-            .finalize()
+        // TODO: Use use VID block commitment.
+        // <https://github.com/EspressoSystems/HotShot/issues/1730>
+        commit::RawCommitmentBuilder::new("Txn Comm").finalize()
```

Comment:
How *does* the transaction commitment relate to the block commitment? Does it at all? My understanding is VID just treats the whole block as a blob, without regard for transaction boundaries. Then I think the only use of the transaction hash would be as a unique index, say for filtering duplicate transactions and stuff, peripheral to the core of consensus? In that case I think this can just be a regular Keccak commitment (ie using the commit library as we normally do)

Reply:
Agree! Ideally, this function could just return the keccak hash, but this would need updates to the `Committable` trait and some other structs that implement `Committable`, so as a short-term fix, I calculated the `Keccak256` hash and then converted it into a `Commitment`.

Reply:
Made an issue for commitment refactoring: https://github.com/EspressoSystems/HotShot/issues/1774.